### PR TITLE
#92 - Better calculation of list of vertices for degenerate centrally symmetric polytopes 

### LINKED ISSUE #92: Faster vertices_list implementation for centrally symmetric polytopes with radius 0
We can just check for radius 0 and only return the center in this case.
This is also an improvement if the radius is 0 only in *some* dimensions.

Types: `Ball1`, `BallInf`, `Hyperrectangle`.

### PR DESCRIPTION
**Pull request by mforets**
Closes #92.

### CODE REVIEW

File: src/AbstractHyperrectangle.jl [hunk 1]
```diff
@@ -51,6 +51,10 @@ For high dimensions, it is preferable to develop a `vertex_iterator` approach.
 """
 function vertices_list(H::AbstractHyperrectangle{N}
                       )::Vector{Vector{N}} where {N<:Real}
+    # fast evaluation if B has radius 0
+    if contains(==, radius_hyperrectangle(H), zero(N))
```
[the remark this answers is not in the document]
Comment (schillic) [reply src/AbstractHyperrectangle.jl]: If this was referring to [this comment](https://github.com/JuliaReach/LazySets.jl/issues/92#issue-282954218), I meant that the result should contain less than 2^n vertices.

File: test/unit_Hyperrectangle.jl [hunk 2]
```diff
@@ -133,4 +133,11 @@ for N in [Float64, Rational{Int}, Float32]
     P = linear_map(Diagonal(N[1., 2., 3., 4.]),
                    Approximations.overapproximate(H1 * H1))
     @test P isa VPolytope # in 4D we get a VPolytope
+
+    # check that vertices_list is computed correctly if the hyperrectangle
+    # is "degenerate" in the sense that its radius is zero in at least one dimension
+    # this test would take very long if all 2^100 vertices are computed (see #92)
+    H = Hyperrectangle(fill(N(1.), 100), [fill(N(2.), 99); 0.])
```
Comment (schillic) [inline test/unit_Hyperrectangle.jl]: Apparently, you need to write `N(0.)` again. Otherwise it will make the whole vector a `Float64`.

**Comment by mforets** [comment]
thank for the feedback!

**Comment by schillic** [comment]
Should we add a unit test for the `Ball1`?

**Pull request merged**